idempotency_key/locks.py

```diff
@@ -0,0 +1,28 @@
+import abc
+import threading
+
+from idempotency_key.utils import get_lock_timeout
+
+
+class IdempotencyKeyLock(object):
+    @abc.abstractmethod
+    def acquire(self, *args, **kwargs) -> bool:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def release(self):
+        raise NotImplementedError()
+
+
+class SingleProcessLock(IdempotencyKeyLock):
+    """
+    Should be used only when there is one process sharing the storage class resource.
+    This uses the built-in python threading module to protect a resource.
+    """
+    storage_lock = threading.Lock()
+
+    def acquire(self, *args, **kwargs) -> bool:
+        return self.storage_lock.acquire(blocking=True, timeout=get_lock_timeout())
```

Comment:
Should `get_lock_timeout` be a function of the base class?

Reply:
It can be but all the other functions that obtain settings from settings.py are under utils so it can be used elsewhere is necessary. Happy to change it though if you think this is a better place.